### PR TITLE
Bump K8S to v1.17.2 with KIND

### PR DESCRIPTION
**Pull request by trozet**
Signed-off-by: Tim Rozet <trozet@redhat.com>

### CODE REVIEW

**Comment by trozet** [comment]
@dcbw PTAL

**Comment by trozet** [comment]
[Fail] [sig-network] EndpointSlice [Feature:EndpointSlice] version v1 [It] should create Endpoints and EndpointSlices for Pods matching a Service 
/home/runner/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/endpointslice.go:358

We are missing support for endpoint slices. Filed: 
https://github.com/ovn-org/ovn-kubernetes/issues/1116

Will resolve #1116 in this PR.

**Comment by trozet** [comment]
Going to just rebase this and skip the failing test for now so we can move to newer stuff.

**Comment by trozet** [comment]
@dcbw @as-com @girishmg PTAL

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func handleFlags() {
 
 var viperConfig = flag.String("viper-config", "", "The name of a viper config file (https://github.com/spf13/viper#what-is-viper). All e2e command line parameters can also be configured in such a file. May contain a path and may or may not contain the file suffix. The default is to look for an optional file with `e2e` as base name. If a file is specified explicitly, it must be present.")
 
-func init() {
+func realInit() {
```
Comment (dave-tucker) [inline test/e2e/e2e_suite_test.go:32]: I think it might be easier to just fold this in to `TestMain`

**Comment by as-com** [comment]
LGTM

File: test/e2e/e2e_suite_test.go (hunk 1 above)
Comment (trozet) [reply test/e2e/e2e_suite_test.go:32]: ok will do

**Comment by nerdalert** [comment]
> I am hitting an issue running tests out of `ovn-org/ovn-kubernetes/test/e2e/e2e_test.go` on this PR. I am hacking up [ginkgo-e2e.sh](https://github.com/kubernetes/kubernetes/blob/master/hack/ginkgo-e2e.sh) to get e2e tests local to ovn-kubernetes to run.
> 
> Trying to isolate the issue I am able to successfully run tests if I rollback the changes to `ovn-kubernetes/test/e2e/go.mod` but leaving everything else the same (k8s version in kind.sh etc). Here is the error I am seeing [e2e-error.log](https://gist.github.com/nerdalert/921e2f20e9d7900aaa12ff7b59c3baec). I also tried different Go versions since that can be problematic with kubetest without any difference. Could just be my environment but wanted to throw it out there in case it's something glaringly obvious. Ty.

I found the issue to be the ginkgo version I was running along with an older Kubernetes branch. Works as advertised. Ty. 

LGTM 🥇

**Pull request merged**